src/frontend/src/binder/relation/table_or_source.rs

```diff
@@ -114,15 +101,93 @@ impl Binder {
+                (
+                    Relation::Source(Box::new(source_catalog.into())),
+                    source_catalog.columns.clone(),
+                )
+            };
+
+            match schema_name {
+                Some(schema_name) => {
+                    let schema_path = SchemaPath::Name(schema_name);
+                    if schema_name == PG_CATALOG_SCHEMA_NAME {
+                        if let Ok(sys_table_catalog) =
+                            catalog.get_sys_table_by_name(db_name, table_name)
+                        {
+                            resolve_sys_table_relation(sys_table_catalog)
+                        } else {
+                            return Err(ErrorCode::NotImplemented(
+                                format!(
+                                    r###"pg_catalog.{} is not supported, please use `SHOW` commands for now.
+`SHOW TABLES`,
+`SHOW MATERIALIZED VIEWS`,
+`DESCRIBE <table>`,
+`SHOW COLUMNS FROM [table]`
+"###,
+                                    table_name
+                                ),
+                                1695.into(),
+                            ).into());
+                        }
+                    } else if let Ok((table_catalog, schema_name)) =
+                        catalog.get_table_by_name(db_name, schema_path, table_name)
+                    {
+                        resolve_table_relation(table_catalog, schema_name)?
+                    } else if let Ok((source_catalog, _)) =
+                        catalog.get_source_by_name(db_name, schema_path, table_name)
+                    {
+                        resolve_source_relation(source_catalog)
+                    } else {
+                        return Err(RwError::from(CatalogError::NotFound(
+                            "table or source",
+                            table_name.to_string(),
+                        )));
+                    }
+                }
+                None => {
+                    let user_name = &self.auth_context.user_name;
+                    let paths = self.search_path.path();
+                    let pg_catalog_position = paths
+                        .iter()
+                        .position(|path| path == PG_CATALOG_SCHEMA_NAME)
+                        .unwrap();
+
+                    let search_path_left =
+                        SearchPath::from_str(&paths[0..pg_catalog_position].join(", "))?;
+                    let schema_path_left = SchemaPath::Path(&search_path_left, user_name);
+                    let search_path_right = SearchPath::from_str(
+                        &paths[pg_catalog_position + 1..paths.len()].join(", "),
+                    )?;
+                    let schema_path_right = SchemaPath::Path(&search_path_right, user_name);
+                    let schema_path = SchemaPath::Path(&self.search_path, user_name);
+
+                    if let Ok((table_catalog, schema_name)) =
```

Comment:
I tested in postgreSQL, system tables have a higher priority when tables with name  like `pg_xxxx` existed in both `pg_catalog` and other schemas in search path:
```
postgres=# create table xxx.pg_index(v int);
CREATE TABLE
postgres=# set search_path to xxx;
SET
postgres=# select * from pg_index;
/// show data in pg_catalog.pg_index
postgres=# select * from xxx.pg_index;
 v
---
(0 rows)
postgres=# \dt pg_index
           List of relations
   Schema   |   Name   | Type  | Owner
------------+----------+-------+--------
 pg_catalog | pg_index | table | august
(1 row)

postgres=# \dt xxx.pg_index
         List of relations
 Schema |   Name   | Type  | Owner
--------+----------+-------+--------
 xxx    | pg_index | table | august
(1 row)
```

And luckily in implementation of this PR left part of search path will be alway empty as I comment in L160. How about simplify these codes as follows:
1. resolve system tables, return if success.
2. resolve tables and sources on search_path.real_path(), return if success.
3. not found.
if so, maybe there's no need to keep hidden `pg_catalog` in search_path IIUC.

Reply:
If you set search_path to `xxx, pg_catalog`, you will get a different answer. So I divide it to two parts.

Reply:
Got it, let's simplify these code as we discussed offline as follows:
```rust
for path in search_path.path() {
    if path == PG_CATALOG_SCHEMA_NAME {
        get_sys_table_by_name(xxx, yyy)
    } else {
        get_table_by_name(xxx, SchemaPath::Name(path), yyy)
    }
}
```